code/modules/mob/living/carbon/alien/diona/diona_powers.dm

```diff
@@ -79,96 +70,92 @@
 			qdel(D)
-			return 1
+			return TRUE
 		else
-
 			D.gestalt = src
 			D.sync_languages(D.gestalt)
 			D.update_verbs()
 			sleep(2)
-			if (is_diona() == DIONA_NYMPH)//We only care about biomass if we're a nymph
+			if(is_diona() == DIONA_NYMPH) //We only care about biomass if we're a nymph
 				src.adjustNutritionLoss(-NYMPH_ABSORB_NUTRITION)
 				src.adjustNutritionLoss(-D.nutrition)
-				D.nutrition = 0
-
-			to_chat(D, "<span class='notice'>You feel your being twine with that of \the [src] as you merge with its biomass.</span>")
-			to_chat(src, "<span class='notice'>You feel your being twine with that of \the [D] as it merges with your biomass.</span>")
+				D.nutrition = FALSE
+			to_chat(D, span("notice", "You feel your being twine with that of \the [src] as you merge with its biomass."))
+			to_chat(src, span("notice", "You feel your being twine with that of \the [D] as it merges with your biomass."))
 			for(var/obj/O in D.contents)
 				D.drop_from_inventory(O)
 			D.forceMove(src)
-
 			D.stat = CONSCIOUS
 			status_flags |= PASSEMOTES
-			return 1
-
+			return TRUE
 	else
-		return 0
+		return FALSE
 
 //Split allows a nymph to peel away from a gestalt and be a lone agent
 /mob/living/carbon/alien/diona/proc/split()
 	set category = "Abilities"
 	set name = "Break from gestalt"
 	set desc = "Split away from your gestalt as a lone nymph."
 
-	if(stat == DEAD || paralysis || weakened || stunned || restrained())
-		return
+	if(use_check_and_message(usr))
+		return FALSE
 
 	if(echo)
-		to_chat(src, "<span class='notice'>Your host is still storing you as a nymphatic husk and preventing your departure.</span>")
+		to_chat(src, span("notice", "Your host is still storing you as a nymphatic husk and preventing your departure."))
 		return
 
-	if(!(istype(src.loc,/mob/living/carbon)))
+	if(!iscarbon(loc))
 		src.verbs -= /mob/living/carbon/alien/diona/proc/split
 		return
 
-	var/r = alert(src,"Splitting will remove you from your gestalt and deposit you on the ground, allowing you to go it alone. If you had any stored biomass before you joined the gestalt, you will not get it back. Are you sure you wish to split?", "Confirm Split", "Time to leaf", "I'll stick around")
-	if (r != "Time to leaf")
+	var/r = alert(src, "Splitting will remove you from your gestalt and deposit you on the ground, allowing you continue alone. If you had any stored biomass before you joined the gestalt, you will not get it back. Are you sure you wish to split?", "Confirm Split", "I am ready to leave.", "I'll stick around.")
+	if(r != "I am ready to leave.")
 		return
 
-	to_chat(src.loc,  "<span class='warning'>You feel a pang of loss as [src] splits away from your biomass.</span>")
-	to_chat(src, "<span class='notice'>You wiggle out of the depths of [src.loc]'s biomass and plop to the ground.</span>")
+	to_chat(src.loc, span("warning", "You feel a pang of loss as [src] splits away from your biomass."))
```

Comment:
```suggestion
	to_chat(src.loc, span("warning", "You feel a pang of loss as [src] splits away from your biomass."))
```

Nymph's loc could potentially change while they deal with the alert. Should send this message to a ref of the host instead of src.loc